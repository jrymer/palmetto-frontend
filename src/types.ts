export interface Weather {
  description: string;
  humidity: string;
  main: string;
  pressure: string;
  temperature: Temperature;
  wind: Wind;
  visibility: string;
}

export interface Temperature {
  feelsLike: string;
  tempActual: string;
  tempMin: string;
  tempMax: string;
}

export interface Wind {
  direction: string;
  speed: string;
}

export interface Time {
  sunrise: number;
  sunset: number;
}

export interface Coordinates {
  lon: number;
  lat: number;
}

export interface WeatherPayload {
  coords: Coordinates;
  time: Time;
  weather: Weather;
  name: string;
}

export type GooglePrediction = google.maps.places.AutocompleteResponse;

export interface PredictionOption {
  display: string;
  value: string;
  id: string;
}

export interface WeatherQueryParams {
  search: {
    city: string;
    coords?: Coordinates;
  };
  unit: Units;
}

export type Units = 'imperial' | 'metric';

export const DEFAULT_CITY_SEARCH = 'Boulder, CO, USA';
