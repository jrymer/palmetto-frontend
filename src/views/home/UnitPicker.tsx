import React from 'react';
import styled from 'styled-components';

import { Units } from '../../types';

const UnitPickerContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: var(--col-width);
  margin: ${(props) => props.theme.margin};
  border: solid thin #ccc;
`;

const UnitPickerItem = styled.div<UnitPickerItemProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  height: -webkit-fill-available;
  width: 100%;
  padding: ${(props) => props.theme.padding};
  background-color: ${(props) => (props.active ? props.theme.colors.primary : '#fff')};
  color: ${(props) => (props.active ? '#fff' : '#000')};

  &:hover {
    filter: brightness(85%);
  }
`;

interface UnitPickerItemProps {
  active: boolean;
}

interface UnitPickerProps {
  unit: Units;
  handleSetUnit: (unit: Units) => void;
}
/**
 *
 *
 * @param {Units} unit Unit that is active
 * @param {(unit: Units) => void} handleSetUnit sets new unit
 *
 * @returns A unit picker component
 */
const UnitPicker: React.FC<UnitPickerProps> = ({ unit, handleSetUnit }) => {
  const handleSelect = (u: Units) => {
    handleSetUnit(u);
  };

  return (
    <UnitPickerContainer>
      <UnitPickerItem active={unit === 'imperial'} onClick={() => handleSelect('imperial')}>
        Imperial
      </UnitPickerItem>
      <UnitPickerItem active={unit === 'metric'} onClick={() => handleSelect('metric')}>
        Metric
      </UnitPickerItem>
    </UnitPickerContainer>
  );
};

export default UnitPicker;
