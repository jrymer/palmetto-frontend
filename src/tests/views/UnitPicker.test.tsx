import React from 'react';

import { Units } from '../../types';
import UnitPicker from '../../views/home/UnitPicker';
import { createSnapshot, fireEvent, render, screen } from '../testWrapper';

const mockHandleSetUnit = jest.fn((unit: Units) => unit);

describe('UnitPicker tests', () => {
  it('UnitPicker snapshot is valid with base props', () => {
    const snapshot = createSnapshot(
      <UnitPicker unit={'imperial'} handleSetUnit={mockHandleSetUnit} />
    );
    expect(snapshot).toMatchSnapshot();
  });
  it('Can click', () => {
    render(<UnitPicker unit={'imperial'} handleSetUnit={mockHandleSetUnit} />);
    const imperial = screen.getByText(/imperial/i);
    const metric = screen.getByText(/metric/i);
    expect(imperial).toBeInTheDocument();
    expect(metric).toBeInTheDocument();
    fireEvent.click(imperial);
    expect(mockHandleSetUnit).toHaveBeenCalled();
  });
});
