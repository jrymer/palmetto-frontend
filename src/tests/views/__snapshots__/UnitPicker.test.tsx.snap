// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitPicker tests UnitPicker snapshot is valid with base props 1`] = `
<div
  className="sc-bczRLJ frTItf"
>
  <div
    className="sc-gsnTZi jYMLFq"
    onClick={[Function]}
  >
    Imperial
  </div>
  <div
    className="sc-gsnTZi jzlcWz"
    onClick={[Function]}
  >
    Metric
  </div>
</div>
`;
